app/src/main/java/lvc/com/movies/ui/listmovies/MoviesAdapter.java

```diff
@@ -0,0 +1,112 @@
+package lvc.com.movies.ui.listmovies;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+
+import java.util.Date;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import lvc.com.movies.R;
+import lvc.com.movies.model.Movie;
+import lvc.com.movies.ui.DiffAdapter;
+import lvc.com.movies.ui.OnItemClickRV;
+import lvc.com.movies.util.ImageURLBuilder;
+import lvc.com.movies.util.Objects;
+
+/**
+ * Created by leonardo2050 on 28/01/18.
+ */
+//
+public class MoviesAdapter extends DiffAdapter<Movie, MoviesAdapter.MovieViewHolder> {
+
+    private OnItemClickRV<Movie> onItemClickRV;
+    private Context context;
+
+    public MoviesAdapter(Context context) {
+        this.context = context;
+    }
+
+    public void setOnItemClickRV(OnItemClickRV<Movie> onItemClickRV) {
+        this.onItemClickRV = onItemClickRV;
+    }
+
+    @Override
+    public MovieViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_movie, parent, false);
+
+        return new MovieViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(final MovieViewHolder holder, int position) {
+        final Movie movie = getItem(position);
+        holder.textViewTitle.setText(movie.getTitle());
+        holder.textViewOverview.setText(toStr(movie.getReleaseDate()));
+        loadImageByUrl(holder.imageViewMovie, movie.getPosterPath());
+
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(onItemClickRV == null) {
+                    return;
+                }
+                onItemClickRV.onItemClick(movie);
+            }
+        });
+    }
+
+    private static String toStr(Date releaseDate) {
+        String dateStr = "";
+        if(releaseDate != null) {
+            dateStr = SimpleDateFormat.getDateInstance().format(releaseDate);
+        }
+        return dateStr;
+    }
+
+    private void loadImageByUrl(ImageView imageView, String path) {
```

Comment:
This method is repeated in `DetailMovieFragment`. How could we remove this repetition? What kind of strategy could we use in order to do it?

Reply:
I could have created a custom ImageView that receives a URL as input and are intelligent  enough to auto load an Image in itself.

Another alternative is, I could create a utility class that holds this logic, and I can use it in different places.  

It should be a good Idea not to use a framework directly like that so it will be easy to replace it in the future if necessary.